iceoryx_utils/include/iceoryx_utils/posix_wrapper/timer.hpp

```diff
@@ -76,6 +76,18 @@ class Timer
         PERIODIC
     };
 
+    /// @brief
+    ///   when the callback is still running when the callback should be retriggered:
+    ///     - SOFT_TIMER = do nothing
+    ///     - ASAP_TIMER = retrigger callback right after the callback is finished
+    ///     - HARD_TIMER = terminate
+    enum class TimerType
+    {
+        SOFT_TIMER,
```

Comment:
it would be nicer if the name would reflect the behaviour, e.g. if the `TimerType` would be a `CatchUpPolicy` we could have `SKIP_TO_NEXT_BEAT`, `IMMEDIATE` and `TERMINATE`